examples/conditional-artifacts.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: conditional-artifacts-
+spec:
+  entrypoint: main
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  templates:
+    - name: main
+      steps:
+        - - name: coinflipmain
+            template: coinflip
+        - - name: printmessage
+            template: printmessage
+            arguments:
+              artifacts:
+                - name: message
+                  from: "{{steps.coinflipmain.outputs.artifacts.step-result}}"
+    
+    - name: coinflip
+      steps:
+        - - name: flipcoin
+            template: flipcoin1
+        - - name: heads
+            template: heads
+            when: "{{steps.flipcoin.outputs.result}} == heads"
+          - name: tails
+            template: tails
+            when: "{{steps.flipcoin.outputs.result}} == tails"
+      outputs:
+        artifacts:
+          - name: step-result
+            fromExpression: "steps.flipcoin.outputs.result == 'heads'?steps.heads.outputs.artifacts.headsresult:steps.tails.outputs.artifacts.tailsresult"
+    
+    - name: flipcoin1
+      script:
+        image: python:alpine3.6
+        command: [python]
+        source: |
+          import random
+          result = "heads" if random.randint(0,1) == 0 else "tails"
+          print(result)
+          
+    - name: heads
+      container:
+        image: alpine:3.6
+        command: [sh, -c]
+        args: ["sleep 40; echo \"it was heads\" > /result.txt"]
+        volumeMounts:
+          - name: workdir
+            mountPath: /mnt/vol
+      outputs:
+        artifacts:
+          - name: headsresult
+            path: /result.txt
+
+            
+    - name: tails
+      container:
+        image: alpine:3.6
```

Comment:
can we always use the standard testing images? I think I would just use `python:3.6` for all tests